### PR TITLE
Change all beans.xml to make bean archives implicit.

### LINKED ISSUE #235: SR CP artifacts are currently transferred into explicit bean archives while they could be implicit
All SR CP artifacts that use `beans.xml` add it as an empty file. This, [according to spec](https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#bean_archive), means they become _explicit bean archives_ with discovery mode `all`. In plain words, every class is scanned and can become a bean.

This doesn't really make sense and consumes resources. We could make all bean archives use `annotated` discovery, meaning that only classes with bean defining annotations can become beans. For this we'd need to change `beans.xml` a little. This shouldn't break any existing setup because all parts of SR CP that we want to be recognized already use those annotations.

Note that this change is for environments using Weld, ArC (Quarkus) doesn't support other than implicit archives and ignores contents of `beans.xml`.

I'll issue a PR momentarily.

### PR DESCRIPTION
**Pull request by manovotn**
Fixes #235

### CODE REVIEW

**Review by FroMage — APPROVED** [review]

**Comment by manovotn** [comment]
Ah, tests seem to rely on that discovery mode, will fix that in a bit.

**Comment by manovotn** [comment]
I have figured one of the test failures, but I can't understand why the `FullStackTest` crashes with changed discovery mode. It just gives me 404 but it has to have something to do with bean discovery.
ATM I have reverted the PR to just keep discovery mode `all` within tests (which is harmless anyway) and I will look into the above described problem tomorrow.

**Comment by manovotn** [comment]
Ok, turns out this is the way it should be. Bean discovery mode `all` is required in tests because of `ResteasyCdiExtension`. Without it, Weld won't fire `ProcessAnnotatedType` for rest resources (because they don't have bean defining annotations) and resteasy won't know about them, hence 404. Subsequently, we won't fire `ProcessInjectionTarget` which they extension is trying to wrap in order to perform their injection, so even if you specify the resource manually, this part will fail because of no injection into the resource.
Long story short, they extension is somewhat weird and relies in CDI mode `all` to actually work its magic. Having this mode in just tests is OK, so we can just keep it that way.

@FroMage please re-stamp the PR whenever you have some spare minute.

**Review by FroMage — APPROVED** [review]
